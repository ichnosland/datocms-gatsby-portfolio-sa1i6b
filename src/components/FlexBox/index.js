/**
*
* FlexBox
*
*/
import styled, { css } from 'styled-components';

import Div from '../Div';
import media from 'style/mediainjector';

export const FlexBox = styled(Div)`
  display: flex;
  flex-direction: ${(props) => props.direction ? props.direction : 'row'};
  justify-content: ${(props) => props.justifyContent ? props.justifyContent : 'center'};
  align-items: ${(props) => props.alignItems ? props.alignItems : 'center'};
  align-content: ${(props) => props.alignContent ? props.alignContent : 'center'};
  flex-wrap: ${(props) => props.wrap ? props.wrap : 'nowrap'};
  ${media.print`
   ${({ noPrint }) => noPrint && css`
       display:none;
   `}
 `}
`;

export const FlexChild = styled(FlexBox)`
  order: ${(props) => props.order};
  flex-grow: ${(props) => props.flexGrow};
  flex-basis: ${(props) => props.flexBasis};
  align-self: ${(props) => props.alignSelf};
`;

export const FlexWrap = styled(FlexBox)`
  position: relative;
  flex-direction: column;
  justify-content: space-between;
  min-height: 100%;
  height: 100%;
  ${media.desktop`
    &:before {
      content: '';
      display: block;
    }
  `}
  ${media.mobile`
    display: block;
  `}
`;

export default FlexBox;
