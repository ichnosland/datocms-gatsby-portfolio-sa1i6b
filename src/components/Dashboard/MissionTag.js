/*
 *
 * MissionTag
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import Div from 'components/Div';
import FlexBox from 'components/FlexBox';
import { colore } from 'style/color';

export const MissionTagBox = styled(FlexBox)`
  width: 25px;
  height: 100%;
  position: absolute;
  left: ${(props) => props.theme.missionTag.left};
  top: ${(props) => props.theme.missionTag.top};
  text-align: center;
  flex-direction: column;
  flex: 1 auto;
  justify-content: center;
  ${({ searchActive }) => searchActive && css`
    display: none;
  `}
`;

MissionTagBox.defaultProps = {
  theme: {
    missionTag: {
      left: '6px',
      top: '10px',
    },
  },
};

export const MissionTagNumber = styled(Div)`
  color: ${(props) => props.color ? props.color : colore.ui.contrast};
  display: inline-block;
  width: 22px;
  margin: 0 auto;
  padding: ${(props) => props.theme.missionTag.padding};
  text-align: center;
  font-weight: 700;
  background-color: ${(props) => props.bgColor ? props.bgColor : props.theme.brand};
  font-size: 11px;
  border-radius: 3px;
`;

MissionTagNumber.defaultProps = {
  theme: {
    brand: '#00BBEF',
    missionTag: {
      padding: '20px',
      top: '10px',
    },
  },
};

class MissionTag extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const {
      number,
      color,
      bgColor,
      searchActive,
    } = this.props;
    return (
      <MissionTagBox
        searchActive={searchActive}
      >
        <MissionTagNumber
          color={color}
          bgColor={bgColor}
        >
          {number}
        </MissionTagNumber>
      </MissionTagBox>
    );
  }
}

MissionTag.propTypes = {
  number: PropTypes.number.isRequired,
  color: PropTypes.string,
  bgColor: PropTypes.string,
  searchActive: PropTypes.bool,
};

export default MissionTag;
