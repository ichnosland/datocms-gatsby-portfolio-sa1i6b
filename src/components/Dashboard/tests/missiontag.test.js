import React from 'react';
import { shallow } from 'enzyme';

import FlexBox from 'components/FlexBox';
import MissionTag, { MissionTagBox, MissionTagNumber } from '../MissionTag';

const number = 3;

describe('<MissionTag />', () => {
  it('should render a MissionTagBox', () => {
    const renderedComponent = shallow(
      <MissionTag number={number} />
    );
    expect(renderedComponent.find(MissionTagBox).length).toEqual(1);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css /w props searchActive', () => {
    const renderedComponent = shallow(
      <MissionTag number={number} searchActive />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<MissionTagBox />', () => {
  it('should render a FlexBox', () => {
    const renderedComponent = shallow(
      <MissionTagBox />
    );
    expect(renderedComponent.find(FlexBox).length).toEqual(1);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css /w props searchActive', () => {
    const renderedComponent = shallow(
      <MissionTagBox searchActive />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<MissionTagNumber />', () => {
  it('should render a Div', () => {
    const renderedComponent = shallow(
      <MissionTagNumber />
    );
    expect(renderedComponent.find('Div').length).toEqual(1);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css /w props color', () => {
    const color = 'red';
    const renderedComponent = shallow(
      <MissionTagNumber color={color} />
    );
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css /w props bgColor', () => {
    const bgColor = 'green';
    const renderedComponent = shallow(
      <MissionTagNumber bgColor={bgColor} />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
