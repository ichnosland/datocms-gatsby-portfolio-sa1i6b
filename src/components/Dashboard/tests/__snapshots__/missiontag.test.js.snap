// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MissionTag /> should render a MissionTagBox 1`] = `
<MissionTag__MissionTagBox
  theme={
    Object {
      "missionTag": Object {
        "left": "6px",
        "top": "10px",
      },
    }
  }
>
  <MissionTag__MissionTagNumber
    theme={
      Object {
        "brand": "#00BBEF",
        "missionTag": Object {
          "padding": "20px",
          "top": "10px",
        },
      }
    }
  >
    3
  </MissionTag__MissionTagNumber>
</MissionTag__MissionTagBox>
`;

exports[`<MissionTag /> should render its css /w props searchActive 1`] = `
<MissionTag__MissionTagBox
  searchActive={true}
  theme={
    Object {
      "missionTag": Object {
        "left": "6px",
        "top": "10px",
      },
    }
  }
>
  <MissionTag__MissionTagNumber
    theme={
      Object {
        "brand": "#00BBEF",
        "missionTag": Object {
          "padding": "20px",
          "top": "10px",
        },
      }
    }
  >
    3
  </MissionTag__MissionTagNumber>
</MissionTag__MissionTagBox>
`;

exports[`<MissionTagBox /> should render a FlexBox 1`] = `
.c0 {
  width: 25px;
  height: 100%;
  position: absolute;
  left: 6px;
  top: 10px;
  text-align: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 auto;
  -ms-flex: 1 auto;
  flex: 1 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<FlexBox
  className="c0"
  theme={
    Object {
      "missionTag": Object {
        "left": "6px",
        "top": "10px",
      },
    }
  }
/>
`;

exports[`<MissionTagBox /> should render its css /w props searchActive 1`] = `
.c0 {
  width: 25px;
  height: 100%;
  position: absolute;
  left: 6px;
  top: 10px;
  text-align: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 auto;
  -ms-flex: 1 auto;
  flex: 1 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: none;
}

<FlexBox
  className="c0"
  searchActive={true}
  theme={
    Object {
      "missionTag": Object {
        "left": "6px",
        "top": "10px",
      },
    }
  }
/>
`;

exports[`<MissionTagNumber /> should render a Div 1`] = `
.c0 {
  color: rgb(255,255,255);
  display: inline-block;
  width: 22px;
  margin: 0 auto;
  padding: 20px;
  text-align: center;
  font-weight: 700;
  background-color: #00BBEF;
  font-size: 11px;
  border-radius: 3px;
}

<Div
  className="c0"
  theme={
    Object {
      "brand": "#00BBEF",
      "missionTag": Object {
        "padding": "20px",
        "top": "10px",
      },
    }
  }
/>
`;

exports[`<MissionTagNumber /> should render its css /w props bgColor 1`] = `
.c0 {
  color: rgb(255,255,255);
  display: inline-block;
  width: 22px;
  margin: 0 auto;
  padding: 20px;
  text-align: center;
  font-weight: 700;
  background-color: green;
  font-size: 11px;
  border-radius: 3px;
}

<Div
  bgColor="green"
  className="c0"
  theme={
    Object {
      "brand": "#00BBEF",
      "missionTag": Object {
        "padding": "20px",
        "top": "10px",
      },
    }
  }
/>
`;

exports[`<MissionTagNumber /> should render its css /w props color 1`] = `
.c0 {
  color: red;
  display: inline-block;
  width: 22px;
  margin: 0 auto;
  padding: 20px;
  text-align: center;
  font-weight: 700;
  background-color: #00BBEF;
  font-size: 11px;
  border-radius: 3px;
}

<Div
  className="c0"
  color="red"
  theme={
    Object {
      "brand": "#00BBEF",
      "missionTag": Object {
        "padding": "20px",
        "top": "10px",
      },
    }
  }
/>
`;
